plugin/path_impersonated_account_secrets.go

```diff
@@ -49,6 +49,9 @@ func (b *backend) pathImpersonatedAccountAccessToken(ctx context.Context, req *l
 	if err != nil {
 		return nil, err
 	}
+	if config == nil {
+		return logical.ErrorResponse("configuration for secrets engine does not exist"), nil
```

Comment:
We don't _need_ to return an error response here. For example, this isn't done in the code that creates service account key secrets [secrets_service_account_key.go#L160](https://github.com/hashicorp/vault-plugin-secrets-gcp/blob/main/plugin/secrets_service_account_key.go#L160). The config is only used for the TTL and MaxTTL in this context. Also, the docs state that you can leave the config unwritten to use application default credentials (see [step 2](https://developer.hashicorp.com/vault/docs/secrets/gcp#setup)).

Looking for thoughts on this!

Reply:
Given there's a valid use case for not writing an engine config, I'd say we probably shouldn't return an error or it inadvertently blocks off that entire path.

Reply:
I definitely agree that if the config is not set this should not error out.

I think it probably makes the most sense to just not pass `Lifetime` with `impersonate.CredentialsTokenSource` and let GCP handle the default case.

Note that this is probably going to be the standard usage by most users.

For those that need to change the default of 1hr, a GCP [org policy constraint](https://cloud.google.com/resource-manager/docs/organization-policy/restricting-service-accounts#extend_oauth_ttl) of `constraints/iam.allowServiceAccountCredentialLifetimeExtension` must be allowed/enabled.

Reply:
I agree, thanks! I've changed this in [bcd7690](https://github.com/hashicorp/vault-plugin-secrets-gcp/pull/160/commits/bcd769056647e44bf3412b08b34632766b4548c5) to not return an error if the config isn't set. I tested that the default TTL will be zero when populating the `Lifetime` field. This results in an access token with the default 1hr TTL.